cli/create.go

```diff
@@ -266,6 +266,21 @@ func createSandbox(ctx context.Context, ociSpec oci.CompatOCISpec, runtimeConfig
 		return vc.Process{}, err
 	}
 
+	// Important to create the network namespace before the sandbox is
+	// created, because it is not responsible for the creation of the
+	// netns if it does not exist.
+	if err := setupNetworkNamespace(&sandboxConfig.NetworkConfig); err != nil {
+		return vc.Process{}, err
+	}
+
+	// Run pre-start OCI hooks.
+	err = enterNetNS(sandboxConfig.NetworkConfig.NetNSPath, func() error {
+		return preStartHooks(ctx, ociSpec, containerID, bundlePath)
```

Comment:
While it can work, I wonder why we are not calling it in `start.go`? After all, they are `preStart` hooks not `preCreate` hooks.

Reply:
@bergwolf This is what the OCI spec calls them. We have simply chosen to call it in create as we need the network hook to be called. @sboeuf and I had a discussion about moving to a hotplug model, but lets address that in a separate PR.

I also believe we need to evaluate if there are certain hooks that need to be run inside the VM.

Reply:
I think it's worth pointing out that both the current code, and this PR are somewhat abusing the standard:

- https://github.com/opencontainers/runtime-spec/blob/master/config.md#prestart

Reply:
@amshinde @sboeuf I think we will need to move it to start.go otherwise vm factory will still be lacking network configurations. But I agree with you that it can be handled in a following up PR.